### PR TITLE
feat: update canvas building blocks

### LINKED ISSUE #30261: Task: Replace Dashboard building block with Sort and filter
https://www.notion.so/appsmith/Replace-Dashboard-building-block-625c5501b248447191c1e39edc0300b0?pvs=4

### PR DESCRIPTION
**Pull request by jacquesikot**
## Description
**Goal**
1. To remove the dashboard canvas starter building block and replace it with the Sort and Filter Table building block from the templates gallery.
2. To replace all canvas starter icons with the corresponding building blocks icons.

**Steps**
- Add a new page to the Starter Templates app called Sort and Filter Table
- Export the existing Sort and Filter Template and import into the new page in Starter Templates using PIE
- Change the copy for title and description in the code
- Take a screenshot of the Sort and Filter table home page, upload to Contentful and update the screenshot URL.
- Download and add new Icons for all 3 canvas starter blocks
- Deploy updated Starter Templates application and upload new JSON to S3 under the same name

#### PR fixes following issue(s)
Fixes #30261 

- New feature (non-breaking change which adds functionality)

- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
  - Updated the naming and descriptions of template page layouts for improved clarity.
  - Reorganized and renamed SVG icons for consistency across the platform.

- **Style**
  - Enhanced the styling of template layout titles and descriptions for better readability.

### CODE REVIEW

File: app/client/src/constants/TemplatesConstants.tsx [hunk 1]
```diff
@@ -61,17 +61,17 @@ export const STARTER_BUILDING_BLOCKS = {
...
       ),
       description: createMessage(
-        STARTER_BUILDING_BLOCKS_LAYOUTS.layouts.dashboard.description,
+        STARTER_BUILDING_BLOCKS_LAYOUTS.layouts.sortFilterTable.description,
       ),
-      icon: <Dashboard />,
+      icon: <SortFilterTable />,
       screenshot:
-        "https://s3.us-east-2.amazonaws.com/template.appsmith.com/canvas-starter-page-layout-dashboard.png",
+        "https://images.ctfassets.net/lpvian6u6i39/55ERsTeUvbAzJVaBBsInZr/0009fee0adb710b91c18a5bdc989deeb/canvas-starter-building-block-sort-filter-table.png?fm=png&q=50",
```
Comment (rahulbarwal) [inline app/client/src/constants/TemplatesConstants.tsx:71]: what cdn is this?

Comment (jacquesikot) [reply app/client/src/constants/TemplatesConstants.tsx:71]: Contentful

**Review by rahulbarwal — APPROVED** [review]

**Review by rahulbarwal — APPROVED** [review]

**Pull request merged**